app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java

```diff
@@ -244,18 +239,32 @@ protected void onCreate(Bundle savedInstanceState) {
         mAssistantHandler = new Handler(mAssistantThread.getLooper());
 
         try {
-            Log.d(TAG, "creating voice hat driver");
-            mVoiceHat = new VoiceHatDriver(I2S_BUS, DAC_TRIGGER_GPIO, AUDIO_FORMAT_STEREO);
-            mVoiceHat.registerAudioInputDriver();
-            mVoiceHat.registerAudioOutputDriver();
-            mButton = new Button(BUTTON_PIN, Button.LogicState.PRESSED_WHEN_LOW);
+            if (AUDIO_USE_I2S_VOICEHAT_IF_AVAILABLE) {
+                PeripheralManagerService pioService = new PeripheralManagerService();
+                List<String> i2sDevices = pioService.getI2sDeviceList();
+                if (i2sDevices.size() > 0) {
+                    try {
+                        Log.d(TAG, "creating voice hat driver");
+                        mVoiceHat = new VoiceHatDriver(
+                                BoardDefaults.getI2SDeviceForVoiceHat(),
```

Comment:
Android framework has to use `CamelCase` even for abbreviation. But we have relaxed style guidelines for our sample.